app/javascript/packages/verify-flow/context/address-verification-method-context.tsx

```diff
@@ -0,0 +1,67 @@
+import { createContext, useState } from 'react';
+import type { ReactNode } from 'react';
+
+/**
+ * Mechanisms by which a user can verify their address.
+ */
+export type AddressVerificationMethod = 'phone' | 'gpo';
+
+/**
+ * Context provider props.
+ */
+interface AddressVerificationMethodContextProviderProps {
+  /**
+   * Optional initial context value.
+   */
+  initialMethod?: AddressVerificationMethod;
+
+  /**
+   * Context children.
+   */
+  children?: ReactNode;
+}
+
+/**
+ * Context value.
+ */
+type AddressVerificationMethodContextValue = [
```

Comment:
what was the reasoning behind making this an array context value instead of an object one? The destructuring to an array surprised me

Reply:
> what was the reasoning behind making this an array context value instead of an object one? The destructuring to an array surprised me

Initially I did make it an object, but felt that the context is essentially equivalent to a `useState` in how it works (current value + setter combo), and it helped avoid a couple extra lines of code. But now that you mention it, I think I'll go back to that approach.

Reply:
Updated in [0b7e8be](https://github.com/18F/identity-idp/pull/6429/commits/0b7e8bea8dd7307d557e14442631370c9b064814).